### PR TITLE
first iteration - ingest account and users

### PR DESCRIPTION
**Pull request by erkangz**

### CODE REVIEW

File: src/provider/fetchBatchOfUsers.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  await Promise.all([
+    userCache.putIds(cachedIds),
+    userCache.putEntries(cacheEntries),
+  ]);
+
+  const finished = resultsCount === 0;
+
+  if (finished) {
+    appendFetchSuccess(cache, `users`);
+  }
```
Comment (aiwilliams) [inline src/provider/fetchBatchOfUsers.ts]: This concept is replaced by a more generalized pattern for storing information about the status of a particular resource : https://github.com/JupiterOne/graph-azure/blob/master/src/azure/fetchBatchOfResources.ts#L63

File: src/types/entities.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  employeeType?: string;
+  employeeId?: string;
+  company?: string;
+  department?: string;
+  jobTitle?: string;
+  suspended?: boolean;
+  mfaExclusion?: boolean;
+  mfaExclusionUntil?: number;
+  mfaConfigured?: boolean;
+}
```
Comment (aiwilliams) [inline src/types/entities.ts]: I'm curious, why not the pattern that forces the converter to provide a value? That looks to be what is done in the `UserEntity` above.

File: src/converters/user.test.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      activated: true,
+      addresses: [],
+      allow_public_key: true,
+      attributes: [],
+      company: "",
+      costCenter: "",
+      department: "",
+      description: "",
+      displayname: "",
+      email: "adam.williams@lifeomic.com",
```
[the remark this answers is not in the document]
Comment (erkangz) [reply src/converters/user.test.ts]: easter egg. you found it

File: src/provider/JumpCloudClient.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+    // p-queue is helpful when all requests are running in a single process, or
+    // there are subsequent processes that are considerate of how many requests
+    // the previous process used, but in the case where API calls are made from
+    // concurrent processes, we're going to need something a bit more
+    // sophisticated.
+    this.requestQueue = new PQueue({
+      interval: 5000,
+      intervalCap: 10,
+    });
```
[the remark this answers is not in the document]
Comment (erkangz) [reply src/provider/JumpCloudClient.ts]: no i took it from another integration

File: src/provider/types/users.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import { JumpCloudObject } from ".";
+
+export interface JumpCloudUser extends JumpCloudObject {
+  email?: string | null;
```
[the remark this answers is not in the document]
Comment (erkangz) [reply src/provider/types/users.ts]: yes it sometimes send null.  and sometimes not there

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,7 @@
+node_modules/
+build/
+coverage/
+dist/
+yarn-error.log
+.env
+.DS_Store
```
[the remark this answers is not in the document]
Comment (erkangz) [reply .gitignore]: i know... but for some reason this global gitignore is not working on my machine

File: src/converters/index.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  EntityFromIntegration,
+  RelationshipFromIntegration,
+} from "@jupiterone/jupiter-managed-integration-sdk";
+
+export * from "./account";
+export * from "./user";
+
+export const consoleBaseUrl = "https://console.jumpcloud.com";
+
+export function createHasRelationships(
```
[the remark this answers is not in the document]
Comment (erkangz) [reply src/converters/index.ts]: maybe have these in the SDK?

File: src/provider/cache.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+export interface DataCache<E, D> {
+  putIds: (ids: string[]) => Promise<void>;
+  getIds: () => Promise<string[]>;
+  getData: (key: string) => Promise<D>;
+  getEntries: (keys: string[]) => Promise<E[]>;
+  putEntries: (entries: E[]) => Promise<void>;
+}
+
+export function createUserCache(
```
[the remark this answers is not in the document]
Comment (erkangz) [reply src/provider/cache.ts]: ok will adopt later

File: src/invocationValidator.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+ * additional information in `context.invocationArgs` should also
+ * validate those properties. It is also helpful to perform authentication with
+ * the provider to ensure that credentials are valid.
+ *
+ * The function will be awaited to support connecting to the provider for this
+ * purpose.
+ *
+ * @param context
+ */
+export default async function invocationValidator(
```
[the remark this answers is not in the document]
Comment (erkangz) [reply src/invocationValidator.ts]: added

**Pull request merged**